src/client/localplayer.cpp

```diff
@@ -228,6 +228,7 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 		node = map->getNode(pp, &is_valid_position);
 		if (is_valid_position) {
 			in_liquid = nodemgr->get(node.getContent()).isLiquid();
+			move_resistance = nodemgr->get(node.getContent()).move_resistance;
 			liquid_viscosity = nodemgr->get(node.getContent()).liquid_viscosity;
```

Comment:
```C++
const ContentFeatures &cf = nodemgr->get(node.getContent());
in_liquid = cf.isLiquid();
move_resistance = cf.move_resistance;
liquid_viscosity = cf.liquid_viscosity;
```
same for the other changes below. Alternatively it would be possible to use a `const ContentFeatures *liquid;` member for LocalPlayer to not copy this information all over again. That however would require more code changes.